lib/worker_status.rb

```diff
@@ -0,0 +1,41 @@
+if Rails.configuration.active_job.queue_adapter == :delayed_job
```

Comment:
Might be cleaner to require or instantiate a different implementation where this is used instead of branching the class definition? (E.g., `worker = DelayedJobWorkerStatus.new` or `worker = SidekiqWorkerStatus.new`)